File: Monsoon Break/chat1.py
t=int(input())

while t:
    n=int(input())
    s=input()
    k=int(input())

    l=w=m=0

    for r in range(n):
        if s[r]=='W':
            w+=1

        while w>k:
            if s[l]=='W':
                w-=1
            l+=1

        m=max(m,r-l+1)

    print(m)
    t-=1

File: Monsoon Break/chat1.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin>>t;

    while(t--){
        int n,k;
        string s;
        cin>>n>>s>>k;

        int l=0,w=0,m=0;

        for(int r=0;r<n;r++){
            if(s[r]=='W')w++;

            while(w>k){
                if(s[l]=='W')w--;
                l++;
            }

            m=max(m,r-l+1);
        }

        cout<<m<<"\n";
    }
}